executor.go

```diff
@@ -146,6 +146,11 @@ func (e *executor) start() {
 						// safety check as it'd be strange bug if this occurred
 						return
 					}
+
+					if j.stopTimeReached() {
+						return
+					}
```

Comment:
In order to cover all of the job types, this check should happen inside the runJob() method of the executor right after it checks if the job or executor contexts are canceled

Reply:
In order to support time mocking, we probably need to move the clockwork object off of the scheduler struct and into the executor struct, so that the clockwork time now value can be passed in here

Reply:
I'll take care of moving the clock. I'm going to open a PR to do that.